dev-packages/cli/src/util/git-util.ts

```diff
@@ -14,74 +14,159 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
+import { resolve } from 'path';
 import * as sh from 'shelljs';
-import { fatalExec, getShellConfig } from './command-util';
-import { LOGGER } from './logger';
-import { validateGitDirectory } from './validation-util';
-
-export function isGithubCLIAuthenticated(): boolean {
-    LOGGER.debug('Verify that Github CLI is installed');
-    fatalExec('which gh', 'Github CLI is not installed!');
-
-    const status = sh.exec('gh auth status', getShellConfig());
-    if (status.code !== 0) {
-        if (status.stderr.includes('You are not logged into any GitHub hosts')) {
-            return false;
-        }
-        throw new Error(status.stderr);
-    }
-    if (!status.stderr.trim().includes('Logged in to github.com')) {
-        LOGGER.debug("No user is logged in for host 'github.com'");
-        return false;
-    }
-    LOGGER.debug('Github CLI is authenticated and ready to use');
-    return true;
+import { getShellConfig } from './command-util';
+
+export function isGitRepository(path?: string): boolean {
+    cdIfPresent(path);
+    return (
+        sh
+            .exec('git rev-parse --is-inside-work-tree', getShellConfig({ silent: true }))
+            .stdout.trim()
+            .toLocaleLowerCase() === 'true'
+    );
 }
 
-export function isGitRepository(path: string): boolean {
-    LOGGER.debug(`Check if the given directory is a git repo: ${path}`);
-    sh.cd(path);
-    return sh.exec('git rev-parse --is-inside-work-tree', getShellConfig()).stdout.trim().toLocaleLowerCase() === 'true';
+export function getGitRoot(path?: string): string {
+    cdIfPresent(path);
+    const fileString = sh.exec('git rev-parse --show-toplevel', getShellConfig()).stdout.trim();
+    return resolve(fileString);
 }
 
 export function hasGitChanges(path?: string): boolean {
-    LOGGER.debug(`Check if the directory has git changes:  ${asDebugArg(path)}`);
+    return getUncommittedChanges(path).length > 0;
+}
+
+/**
+ * Returns the files that have uncommitted changes (staged, not staged and untracked) of a git repository.
+ * Filepaths are absolute.
+ */
+export function getUncommittedChanges(path?: string): string[] {
     cdIfPresent(path);
-    return sh.exec('git status --porcelain').stdout.trim().length !== 0;
+    return (
+        sh
```

Comment:
Odd formatting and unnecessary parenthesis, I believe. Was that formatted like this on purpose?

Reply:
That's prettier autoformatting. I guess the inline comments messes up the formatting. It's a little bit better if I move the comment inside of the map function